apps/src/templates/sectionProgressV2/floatingScrollbar/FloatingScrollbar.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import styles from './floating-scrollbar.module.scss';
+
+/**
+ * A component that adds a horizontal scrollbar to the bottom of {children} element.
+ * Scrollbar will appear at the bottom of {child} if the bottom is visible.
+ * Scrollbar will at the bottom of the screen if the bottom of {child} is not visible.
+ *
+ * {children} must:
+ *    be a single element
+ *    be a container with overflow-x: hidden
+ *    have a single contents component that you would like to horizontally scroll
+ * setOnScroll - a function that will be called with an onScroll event.
+ *    This should be added to {child} element as the `onScroll` prop
+ * childContainerRef - a ref to {child} element.
+ *    Set by calling `ref={childContainerRef}` on the {child} element
+ * childContentsRef - a ref to the contents of {child} element.
+ *    Set by calling `ref={childContentsRef}` on the contents of {child} element
+ */
+export default function FloatingScrollbar({
+  children,
+  setOnScroll,
+  childContainerRef,
+  childContentsRef,
+}) {
+  const scrollRef = React.useRef();
+
+  const [childScrollWidth, setChildScrollWidth] = React.useState(0);
+  const [childWidth, setChildWidth] = React.useState(0);
+  const [scrollVisible, setScrollVisible] = React.useState(true);
+
+  if (children.length > 1) {
+    throw new Error('FloatingScrollbar only supports a single child');
+  }
+
+  const childContainerResizeObserver = React.useMemo(
+    () =>
+      new ResizeObserver(([entry]) => {
+        if (entry.borderBoxSize) {
+          setChildWidth(entry.borderBoxSize[0].inlineSize);
+        }
+      }),
+    [setChildWidth]
+  );
+
+  const childContentsResizeObserver = React.useMemo(
+    () =>
+      new ResizeObserver(([entry]) => {
+        if (entry.borderBoxSize) {
+          setChildScrollWidth(entry.borderBoxSize[0].inlineSize + 1);
```

Comment:
What's the effect of the `+ 1` here? Would a different number need to be used for different border widths?

Reply:
I'm not entirely sure, because it's surprisingly hard to test these off by ones, but I am pretty sure it's because of this -1 margin-right: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/sectionProgressV2/progress-table-v2.module.scss#L127